### PR TITLE
Fixing publish test results in gradle task

### PR DESCRIPTION
**Pull request by prawalagarwal**
Description: Currently, gradle task does not publish test results even if one of the tests fail because of the way handler is implemented. This fix will call publish in all cases.

Testing : Manual

### CODE REVIEW

**Comment by DavidStaheli** [comment]
Hi @prawalagarwal, thanks for this important fix!  Do you plan to also port it to the releases/m105 branch so that it makes it into Dev15 RTW?  Please let me know if I can assist with that.  Thanks again.

**Comment by prawalagarwal** [comment]
Hi @DavidStaheli , yes we plan to port it to releases/m105 for RTW. Should I raise a PR for it now?

File: Tasks/Gradle/gradletask.ts [hunk 1]
```diff
@@ -96,25 +96,35 @@ async function execBuild() {
...
-            } 
+            tl.debug(`Gradle result: ${gradleResult}`);
+            return "Success";            
         })
         .fail(function (err) {
             console.error(err);
             tl.debug('taskRunner fail');
-            tl.setResult(tl.TaskResult.Failed, err); 
+            gradleResult = -1;
+            return err;
```
Comment (nigurr) [inline Tasks/Gradle/gradletask.ts]: is it supposed to return promise or plain value?

File: Tasks/Gradle/gradletask.ts [hunk 2]
```diff
@@ -96,25 +96,35 @@ async function execBuild() {
...
+            return "Success";            
         })
         .fail(function (err) {
             console.error(err);
             tl.debug('taskRunner fail');
-            tl.setResult(tl.TaskResult.Failed, err); 
+            gradleResult = -1;
+            return err;            
+        })
+        .then( function(resp) {
```
Comment (nigurr) [inline Tasks/Gradle/gradletask.ts]: indention

**Comment by DavidStaheli** [comment]
Thanks @prawalagarwal.  As soon as you're comfortable merging this PR, feel free to create a new PR for **releases/m105**.

File: Tasks/Gradle/gradletask.ts (hunk 2 above)
Comment (prawalagarwal) [inline Tasks/Gradle/gradletask.ts]: Done. Thanks

File: Tasks/Gradle/gradletask.ts (hunk 1 above)
Comment (prawalagarwal) [inline Tasks/Gradle/gradletask.ts]: promise. fixed.

**Comment by nigurr** [comment]
LGTM

**Pull request merged**